src/Sulu/Bundle/AdminBundle/Resources/js/index.js

```diff
@@ -310,6 +310,16 @@ function processConfig(config: Object) {
 }
 
 function startAdmin() {
+    if (Config.suluVersion !== SULU_ADMIN_BUILD_VERSION) {
```

Comment:
I guess now every developer always gets error messages like this:

```
Sulu administration interface: JavaScript build version mismatch
JavaScript build of the Sulu administration interface does not match the version of the Sulu backend.
Backend version: _._._, JavaScript build version: undefined

Have you forgotten to update the build while upgrading your application?
https://docs.sulu.io/en/latest/upgrades/upgrade-2.x.html
```

Also, the `composer.lock` file is probably not up to date, if I switch the branch locally, which might lead to false positives. Wondering if we can't implement that more gracefully...

Options that come to my mind:

- Ignore mismatch when one of both operands does not seem to be a version
- Ignore mismatch when webpack runs in development mode
- Maybe only show a warning instead of an error? (Might still work, but not sure if people will read the warning)

Thinking about the last point, I am wondering if (especially) PHP developers look at the console tab in the dev tools of the browser. I often have the feeling they don't care about what's written there... That being said, maybe we can move that check somehow to the server side? Although that might be even less graceful...

Reply:
> Ignore mismatch when one of both operands does not seem to be a version

Think I would rather use `_._._` as default value. to solve this That would be consistent to the `SuluVersionPass`.

> Ignore mismatch when webpack runs in development mode

Not sure about this, as the mechanism will not work anymore if somebody commits a watch build by mistake. I think this happens quite often unfortunately 🙈 

> Maybe only show a warning instead of an error? (Might still work, but not sure if people will read the warning)

I used a warning initially. Bun I dont think people will pay attention to thatunfortunatelyy 🙈 

> Thinking about the last point, I am wondering if (especially) PHP developers look at the console tab in the dev tools of the browser.

Probably not very often. But I hope that some of them will have a look at the console if something does not work in the frontend 🙂

Reply:
Changing the default to `_._._` would probably be the minimum we'd have to do... Maybe let's just try that, and see how it behaves then.

I could still imagine that false positives appear during development. I think we have to be careful that this error is not shown too often, otherwise it will just be accepted that it is there :see_no_evil: Therefore I would tend to show it too less than too often.

Reply:
Changed the default value 🙂  

If this matching-strategy proves to be working, we could still add additional logs in the backend in the future. Maybe it would also make sense to display an icon beside the version in the administration interface. But I think I would not yet do this in this PR.